### PR TITLE
TST/REF: de-duplicate SetitemCastingEquivalents

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/series/indexing/test_setitem.py [hunk 1]
```diff
@@ -309,52 +307,59 @@ class SetitemCastingEquivalents:
...
         obj = obj.copy()
+        arr = obj._values
+
         indexer(obj)[key] = val
         tm.assert_series_equal(obj, expected)
 
-    def test_int_key(self, obj, key, expected, val, indexer_sli):
+        self._check_inplace(is_inplace, orig, arr, obj)
+
+    def _check_inplace(self, is_inplace, orig, arr, obj):
```
Comment (jreback) [inline pandas/tests/series/indexing/test_setitem.py:320]: maybe can be more generally located?

**Comment by jreback** [comment]
needs a rebase

**Comment by jbrockmendel** [comment]
rebased + green

File: pandas/tests/series/indexing/test_setitem.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/series/indexing/test_setitem.py:327]: im optimistic we can eventually use this or something like it for a ton of indexing tests, not sure where this particular method/class is going to end up

**Pull request merged**